models/models_methods.go

```diff
@@ -392,33 +392,62 @@ func validateStatusReportName(statusReportName string) error {
 }
 
 func (app *AppModel) Validate() error {
+	return app.internalValidation(true)
+}
+
+func (app *AppModel) MinimalValidation() error {
+	return app.internalValidation(false)
+}
+
+func (app *AppModel) internalValidation(full bool) error {
 	for _, env := range app.Environments {
 		if err := env.Validate(); err != nil {
 			return err
 		}
 	}
+
+	if !full {
+		return nil
+	}
+
 	return validateStatusReportName(app.StatusReportName)
 }
 
 func (config *BitriseDataModel) Validate() ([]string, error) {
+	return config.internalValidation(true)
+}
+
+func (config *BitriseDataModel) MinimalValidation() ([]string, error) {
+	return config.internalValidation(false)
+}
+
+func (config *BitriseDataModel) internalValidation(full bool) ([]string, error) {
```

Comment:
I kept the order of the field validation and did not group the ones which should be performed only in full mode under a big `if` statement. The reason is that the collected warnings with this ordering follows the the order of the fields appearing in the config yaml.